### ISSUE TITLE
Output of networks in Mixture of Experts

### ISSUE THREAD

**Issue by Daemonstool** [itdxer/neupy#175]
I am trying to implement a Mixture of MLP experts. I need one manager that controls n-MLPs. The MLPs each accept 6 inputs and output 6 inputs since I have to deal with six-dimensional data. However when I feed the MLPs to the Mixture of Experts algorithm it asserts that the output of each network should be of size 1. Why am I not allowed to use networks that return more than 1 unit?

```python
insize, outsize = (6, 6)
networks = [
    algorithms.Momentum(
    [
        layers.Input(insize),
        layers.Relu(500),
        layers.Relu(300),
        layers.Linear(outsize),
    ]),
    algorithms.Momentum(
    [
        layers.Input(insize),
        layers.Relu(500),
        layers.Relu(300),
        layers.Linear(outsize),
    ]),
]
n_networks = len(networks)

print(n_networks)

moe = algorithms.MixtureOfExperts(
    networks=networks,
    gating_network=algorithms.GradientDescent(
        layers.Input(insize) > layers.Softmax(n_networks),
        step=0.1,
        verbose=False,
    )
)

**Comment by itdxer** [comment]
@Daemonstool It works in this way, because current implementation was mostly focused on regression tasks with single output. In addition, this implementation applies weighted average where weights defined by the gating network.

I'll consider to update this algorithm in the next release so that it will work for a wider variety of tasks. Thank you for pointing this out.

**Comment by Daemonstool** [comment]
Thank you for your reply. For now I decided to implement this functionality myself in Keras. Good luck with any future work on Neupy :)

**Comment by itdxer** [comment]
Mixture of experts functionality was updated and modified in order to support arbitrary output shapes (even multi-dimensional). Feature will be included in neupy 0.5.3
